components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -9090,4 +8976,86 @@ protected String createAPIRevisionRegistryArtifacts(String apiUUID, int revision
     public List<APIRevision> getAPIRevisions(String apiUUID) throws APIManagementException {
         return apiMgtDAO.getRevisionsListByAPIUUID(apiUUID);
     }
+
+    /**
+     * Adds a new APIRevisionDeployment to an existing API
+     *
+     * @param apiId API UUID
+     * @param apiRevisionId API Revision UUID
+     * @param apiRevisionDeployments List of APIRevisionDeployment objects
+     * @throws APIManagementException if failed to add APIRevision
+     */
+    @Override
+    public void addAPIRevisionDeployment(String apiId, String apiRevisionId, List<APIRevisionDeployment> apiRevisionDeployments) throws APIManagementException {
+        APIIdentifier apiIdentifier = APIUtil.getAPIIdentifierFromUUID(apiId);
+        if (apiIdentifier == null) {
+            throw new APIMgtResourceNotFoundException("Couldn't retrieve existing API with API UUID: "
+                    + apiId, ExceptionCodes.from(ExceptionCodes.API_NOT_FOUND, apiId));
+        }
+        APIRevision apiRevision = apiMgtDAO.getRevisionByRevisionUUID(apiRevisionId);
+        if (apiRevision == null) {
+            throw new APIMgtResourceNotFoundException("Couldn't retrieve existing API Revision with Revision UUID: "
+                    + apiRevisionId, ExceptionCodes.from(ExceptionCodes.API_NOT_FOUND, apiRevisionId));
+        }
+        APITemplateBuilder builder = null;
+        APIGatewayManager gatewayManager = APIGatewayManager.getInstance();
+        APIIdentifier revisionApiIdentifier = getLightweightAPIByUUID(apiRevisionId, tenantDomain).getId();
+        API api = getRevisionAPI(revisionApiIdentifier, apiRevision);
+        api.setSwaggerDefinition(getOpenAPIDefinition(revisionApiIdentifier));
```

Comment:
lets put a if check here to check for swagger def in the returned api obj. if definition is there, no need to get the def again.

Reply:
resolved, moved to getRevisionAPI